### PR TITLE
tests/lib/nested.sh: allow tests to use their own core18 in extra-snaps-path

### PR DESCRIPTION
**Pull request by anonymouse64**
I needed to do this for a nested uc18 test and it failed since the core18 I put
in extra-snaps-path was in conflict with always putting the repacked core18
into $EXTRA_FUNDAMENTAL which leads snap prepare-image to complain about a
duplicated core18 snap to be provided.

### CODE REVIEW

**Review by mardy — COMMENTED** [review]
Just a minor issue. It would also be nice to document how one can provide one's own core18 snap (I guess it's by adding `--snap my-core-18.snap` to `EXTRA_FUNDAMENTAL`, but better be explicit).

File: tests/lib/nested.sh [hunk 1]
```diff
@@ -606,11 +606,19 @@ nested_create_core_vm() {
...
+                    if [ -d "$(nested_get_extra_snaps_path)" ]; then
+                        CORE18_SNAP=$(find extra-snaps -name 'core18*.snap')
+                    fi
+                    if [ -z "$CORE18_SNAP" ]; then
+                        snap download --channel="$CORE_CHANNEL" --basename=core18 core18
+                        repack_core_snap_with_tweaks "core18.snap" "new-core18.snap"
+                        EXTRA_FUNDAMENTAL="$EXTRA_FUNDAMENTAL --snap $PWD/new-core18.snap"
 
-                    repack_core_snap_with_tweaks "core18.snap" "new-core18.snap"
+                        repack_core_snap_with_tweaks "core18.snap" "new-core18.snap"
```
[the remark this answers is not in the document]
Comment (anonymouse64) [reply tests/lib/nested.sh:620]: ah yes thanks to suggesting this, not sure how I ended up with the line duplicated but that was not the intention :-)

[the remark this answers is not in the document]
Comment (anonymouse64) [reply tests/lib/nested.sh:620]: oh actually this line is already duplicated in master it seems :vampire:

[the remark this answers is not in the document]
Comment (anonymouse64) [reply tests/lib/nested.sh]: should be good now, thanks folks

**Review by mardy — APPROVED** [review]
LGTM!

**Review by stolowski — APPROVED** [review]
LGTM, would be good to get +1 from @sergiocazzolato

**Pull request merged**